Sources/NIOHTTPClient/SwiftNIOHTTP.swift

```diff
@@ -128,6 +131,30 @@ public class HTTPClient {
 
         let task = Task(future: promise.futureResult)
 
+        let connection = pool.getConnection(for: request)
+        
+        connection.flatMap { connection -> EventLoopFuture<()> in
+            let channel = connection.channel
+            let addedFuture: EventLoopFuture<Void>
+            if let readTimeout = timeout.read {
+                addedFuture = channel.pipeline.addHandler(IdleStateHandler(readTimeout: readTimeout))
+            } else {
+                addedFuture = channel.eventLoop.makeSucceededFuture(())
+            }
+            return addedFuture.flatMap {
+                let taskHandler = TaskHandler(task: task, delegate: delegate, promise: promise, redirectHandler: redirectHandler)
+                task.setChannel(channel)
+                return channel.pipeline.addHandler(taskHandler)
+            }.flatMap {
+                return channel.writeAndFlush(request)
+            }
+            }.whenFailure { error in
```

Comment:
This indentation is not right.

Reply:
Looks like Xcode auto indent had issues there, I'll se if swiftformat fixes it and will correct it manually otherwise.